packages/ui/react-ui-editor/src/extensions/comments.ts

```diff
@@ -145,17 +164,33 @@ const commentsDecorations = EditorView.decorations.compute([commentsState], (sta
         return undefined;
       }
 
-      if (comment.comment.id === current) {
-        return commentCurrentMark.range(range.from, range.to);
-      } else {
-        return commentMark.range(range.from, range.to);
-      }
+      const mark = createCommentMark(comment.comment.id, comment.comment.id === current);
+      return mark.range(range.from, range.to);
     })
     .filter(nonNullable);
 
   return Decoration.set(decorations);
 });
 
+const commentClickedEffect = StateEffect.define<string>();
+
+const handleCommentClick = EditorView.domEventHandlers({
+  click: (event, view) => {
+    let target = event.target as HTMLElement;
+    // Traverse up the DOM tree looking for an element with data-comment-id
+    while (target && !target.hasAttribute('data-comment-id')) {
```

Comment:
will this limit itself to codemirror or would it traverse up the whole page if no comments exist?

Reply:
It will go all the way up to the html root. It's only on click events though so I'm not too bothered. Can implement a short circuit if you feel it's good to isolate it from the wider app?

Reply:
given the potentially dynamic nature of plugins it feels like it would be safer to be defensive and try to keep things from having unintended side-effects (i.e. another plugin having a `data-comment-id` attribute that this somehow finds triggers clicks against)

Reply:
if it's a huge bother to find get the codemirror root here then I'd just leave it as a todo

Reply:
Looks like we can just use `view.dom` as a reference to the root element.